### PR TITLE
465 feature make time selection modal look better

### PR DESCRIPTION
**Pull request by liviawangg**
### **Changes I have made:**
- Made DesktopDatePicker and DesktopTimePicker a component
- Moved the two components out of BuildingDrawer and next to the search button instead

Desktop view:
- Filter and Sort button to the left of search and Date and Time picker to the right of search bar

Tablet view:
- Filter & Sort button are adjacent to Date & Time Picker, all under search bar

Mobile view:
- Search bar on top, Filter & Sort buttons are under, Date & Time picker last

### **How to view:**
- Once on browsing rooms page, inspect page and drag the responsive view 
- Click on a building to open drawer

### 
Desktop view:
Screenshot 2024-06-20 at 9 36 16 PM
Tablet view:
Screenshot 2024-06-20 at 9 36 37 PM
Phone view:
Screenshot 2024-06-20 at 9 36 54 PM
Building Drawer:
Screenshot 2024-06-20 at 9 37 30 PM

### CODE REVIEW

File: frontend/__tests__/BrowsingPage.test.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+describe("Browsing Page", () => {
+  it("renders DesktopTimePicker", () => {
+    render(
+      <Provider store={store}>
+        <Page />
+      </Provider>
+    );
+    const timePicker = screen.getByRole("button", {
+      name: /Choose time, selected time is/i,
```
Comment (JessicaF) [inline frontend/__tests__/BrowsingPage.test.tsx:17]: I can't find in the code where there is a button with this name - where did you set it? 

Also would be good to also include a test to see if the DatePicker shows up

File: frontend/components/DatePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+"use client";
```
Comment (JessicaF) [inline frontend/components/DatePicker.tsx:1]: The DatePicker and TimePicker have a grey background for me - do you think there's a way for you to fix this? If it takes too long don't worry about it!

**Review by JessicaF — COMMENTED** [review]
Really good overall!!! Thanks so much for implementing the changes 🤩🤩 I've just left a couple comments but if you'd prefer to focus on the Rating System tasks then just leave a comment and I'll fix what I can!

File: frontend/components/DatePicker.tsx (hunk 2 above)
Comment (liviawangg) [reply frontend/components/DatePicker.tsx:1]: yeah... idk how to fix that lol T_T I tried to add an orange broder, but for some reason when I replaced it, the grey border stayed there. and at that point i just gave up lol

Comment (JessicaF) [reply frontend/components/DatePicker.tsx:1]: That's okay! We can leave this for now - have you been able to resolve the other comments?

**Comment by JessicaF** [comment]
@liviawangg Is this ready for review yet! 👀

**Comment by liviawangg** [comment]
> @liviawangg Is this ready for review yet! 👀

yesss :>>

**Review by JessicaF — APPROVED** [review]

**Pull request merged**